lib/validate-schemas.spec.js

```diff
@@ -24,19 +24,157 @@ function loadRemoteSchema(uri) {
     });
 }
 
+const dataKeyTopicRegex = /^[a-zA-Z]([a-zA-Z0-9]+|\.|_)*[a-zA-Z0-9]$/;
+const dataKeyTopicMinLength = 0;
+const dataKeyTopicMaxLength = 64;
+const validatePropertyName = (errors, dataKey) => {
+  if (dataKey.length < dataKeyTopicMinLength) {
+    errors.push(`dataKey '${dataKey}' too short`);
+  }
+
+  if (dataKey.length > dataKeyTopicMaxLength) {
+    errors.push(`dataKey '${dataKey}' too long`);
+  }
+
+  if (!dataKeyTopicRegex.test(dataKey)) {
+    errors.push(`dataKey '${dataKey}' does not match convention`);
+  }
+};
+
+const validateTopic = (filepath, errors, topic) => {
+  if (topic === undefined) {
+    errors.push(`topic not set for ${filepath}`);
+    return;
+  }
+
+  if (topic.length < dataKeyTopicMinLength) {
+    errors.push(`topic '${topic}' too short`);
+  }
+
+  if (topic.length > dataKeyTopicMaxLength) {
+    errors.push(`topic '${topic}' too long`);
+  }
+
+  if (!dataKeyTopicRegex.test(topic)) {
+    errors.push(`topic '${topic}' does not match convention`);
+  }
+};
+
+const validateProcessingType = (filepath, errors, processingType) => {
+  if (processingType === undefined) {
+    errors.push(`processingType not set for ${filepath}`);
+    return;
+  }
+
+  if (
+    processingType !== "uplink_decoder" &&
+    processingType !== "downlink_encoder"
+  ) {
+    errors.push(
+      `processingType not one of ['uplink_decoder', 'downlink_encoder' for ${filepath}`,
+    );
+  }
+};
+
+const validateSchemaProperties = (errors, schema) => {
+  if (schema === undefined) {
+    return;
+  }
+
+  const schemaKeys = Object.keys(schema);
+  const allowedProperties = [
+    "unit",
+    "$ref",
+    "title",
+    "type",
+    "description",
+    "properties",
+    "required",
+    "examples",
+    "hideFromKpis",
+    "minimum",
+    "maximum",
+    "pattern",
+    "uniqueItems",
+    "default",
+    "enum",
+    "items",
+  ];
+
+  const notAllowedProperties = schemaKeys.filter(
+    (property) => !allowedProperties.includes(property),
+  );
+  if (notAllowedProperties.length > 0) {
+    errors.push(
+      notAllowedProperties.map((prop) => `${prop} is not allowed in schema`),
+    );
+  }
+
+  if (schema.properties !== undefined) {
+    Object.keys(schema.properties).forEach((property) => {
+      validatePropertyName(errors, property);
+      validateSchemaProperties(errors, schema.properties[property]);
+    });
+  }
+};
+
+const validateSchemaKeys = (errors, schema) => {
+  if (schema === undefined) {
+    return;
+  }
+
+  const schemaKeys = Object.keys(schema);
+  const allowedProperties = [
```

Comment:
How do we deal with the schemas defined in https://github.com/akenza-io/device-type-library/tree/master/data-models ?

We probably want to support there all possible properties' fields as well as `$id` and `$schema`.